vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -1094,24 +1116,31 @@ def _prepare_decode(
         for req_id in req_ids:
             # adding new blocks if needed
             if self.tkv % self.block_size == 0:
-                self.req_ids2blocks[req_id].append(self.block_pool.popleft())
-            n_blocks = max(n_blocks, len(self.req_ids2blocks[req_id]))
+                req_state: SamplingRequestState = self.requests[req_id]
+                total_tokens = req_state.left_padding % self.block_size \
+                    + req_state.num_computed_tokens + 1
```

Comment:
Could we add a comment to explain this formula (especially the `req_state.left_padding % self.block_size` bit)?